flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/MultipleInputITCase.scala

```diff
@@ -182,6 +182,27 @@ class MultipleInputITCase(shuffleMode: String) extends BatchTestBase {
     )
   }
 
+  @Test
+  def testManyInputs(): Unit = {
+    val rowType = new RowTypeInfo(INT_TYPE_INFO, STRING_TYPE_INFO)
+    val data = Seq(BatchTestBase.row(1, "test"))
+    val nullables: Array[Boolean] = Array(true, true)
+    registerCollection("left_table", data, rowType, "a, b", nullables)
+    registerCollection("right_table", data, rowType, "c, d", nullables)
+
+    val numInputs = 56
```

Comment:
why do we have `56` and not `MAX_SUPPORTED_INPUT_COUNT`?

Reply:
Because the example provided by the user has 56 inputs (after a second check I discover that this should be 55 if we want to follow the user's example). However `MAX_SUPPORTED_INPUT_COUNT` is indeed more reasonable.

Reply:
Oh, I think I now fully understand it. There were two bugs:

1. The limit should be 63 instead of 64
2. And on top of that, `1 << inputCount` was causing that the code would fail already with 32 inputs.

2. Is the reason why the user's query with 56 inputs was failing?

Reply:
Yes. Exactly.